Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.ControlBindingsCollection.cs

```diff
@@ -0,0 +1,185 @@
+    /// </summary>
+    /// 
+    /// <returns>
+    /// The newly created <see cref="T:System.Windows.Forms.Binding"/>.
+    /// </returns>
+    /// <param name="propertyName">The name of the control property to bind. </param><param name="dataSource">An <see cref="T:System.Object"/> representing the data source. </param><param name="dataMember">The property or list to bind to.</param><param name="formattingEnabled">true to format the displayed data; otherwise, false.</param><param name="updateMode">One of the <see cref="T:System.Windows.Forms.DataSourceUpdateMode"/> values.</param><param name="nullValue">When the data source has this value, the bound property is set to <see cref="T:System.DBNull"/>. </param><param name="formatString">One or more format specifier characters that indicate how a value is to be displayed</param><param name="formatInfo">An implementation of <see cref="T:System.IFormatProvider"/> to override default formatting behavior.</param><exception cref="T:System.ArgumentException">The property given by <paramref name="propertyName"/> does not exist on the control or is read-only.-or-The specified data member does not exist on the data source.-or-The data source, data member, or control property specified are associated with another binding in the collection.</exception>
+    public Binding Add(string propertyName, object dataSource, string dataMember, bool formattingEnabled, DataSourceUpdateMode updateMode, object nullValue, string formatString, IFormatProvider formatInfo)
+    {
+        Contract.Requires(dataSource != null);
+        return default(Binding);
+    }
+        
+    // <summary>
+    // Clears the collection of any bindings.
+    // </summary>
+    // public new void Clear();
+        
+    /// <summary>
+    /// Deletes the specified <see cref="T:System.Windows.Forms.Binding"/> from the collection.
+    /// </summary>
+    /// <param name="binding">The <see cref="T:System.Windows.Forms.Binding"/> to remove. </param><exception cref="T:System.NullReferenceException">The <paramref name="binding"/> is null. </exception><filterpriority>1</filterpriority>
+    public new void Remove(Binding binding)
+    {
+        Contract.Requires(binding != null);
+    }
+
+    /// <summary>
+    /// Deletes the <see cref="T:System.Windows.Forms.Binding"/> at the specified index.
+    /// </summary>
+    /// <param name="index">The zero-based index of the item to remove. </param><exception cref="T:System.ArgumentOutOfRangeException">The <paramref name="index"/> value is less than 0, or it is greater than the number of bindings in the collection. </exception><filterpriority>1</filterpriority>
+    public new void RemoveAt(int index)
+    {
+        Contract.Requires(index >= 0);
+    }
```

Comment:
This should probably have `Contract.Requires(index < this.Count);`. The documentation says it throws ArgumentOutOfRangeException if index value is less than 0, or it is greater than the number of bindings in the collection, so we should try to reflect that.

>